### PR TITLE
feat: [SFEQS-1543] Add APIM policy to `getSignerByFiscalCode`

### PR DESCRIPTION
**Pull request by grausof**
Adds a policy to the `getSignerByFiscalCode` endpoint to filter requests based on authorized fiscal codes. In particular, if the issuer uses a test environment, it can contact the `getSignerByFiscalCode` endpoint only with allowed fiscal code.

### List of changes
- Added `io-sign-cosmosdb-issuer-issuers-name` named value
- Added custom policy to `getSignerByFiscalCode`

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

File: src/domains/sign/api/issuer/v1/get_signer_by_fiscal_code_policy/policy.xml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+                <set-variable name="issuerWhitelistFiscalCodes" value="@(((IResponse)context.Variables["response"]).Body.As<JObject>())" />
+                <!-- Cache the query result for 60s to reduce the number of calls to cosmosDb -->
+                <cache-store-value key="@(context.Subscription.Id+"_whitelist_cf")" value="@((JObject)context.Variables["issuerWhitelistFiscalCodes"])" duration="60" />
+            </when>
+        </choose>
+        <cache-lookup-value key="@(context.Subscription.Id+"_issuer_environment")"  variable-name="issuerEnvironment" />
+        <choose>
+            <when condition="@(!context.Variables.ContainsKey("issuerEnvironment"))">
+                <set-variable name="requestDateString" value="@(DateTime.UtcNow.ToString("r"))" />
+                <send-request mode="new" response-variable-name="response" timeout="10" ignore-error="false">
```
Comment (gunzip) [inline src/domains/sign/api/issuer/v1/get_signer_by_fiscal_code_policy/policy.xml:64]:
```suggestion
                <send-request mode="new" response-variable-name="response" timeout="1" ignore-error="false">
```

**Review by gunzip — APPROVED** [review]

**Review by gunzip — APPROVED** [review]
do we get some tracing in case something goes wrong?

File: src/domains/sign/apim.tf [hunk 2]
```diff
@@ -91,6 +101,15 @@ module "apim_io_sign_product" {
   policy_xml = file("./api_product/sign/_base_policy.xml")
 }
 
+resource "azurerm_api_management_api_operation_policy" "get_signer_by_fiscal_code_policy" {
+  api_name            = "io-sign-api"
```
Comment (manuelcoppotelli) [inline src/domains/sign/apim.tf:105]:
```suggestion
  api_name            = module.apim_io_sign_issuer_api_v1.name
```

Two birds with one stone:
- we make sure to set the correct API name
- we define a dipendency between the two resoruces

**Review by manuelcoppotelli — CHANGES_REQUESTED** [review]

File: src/domains/sign/apim.tf (hunk 2 above)
Comment (grausof) [reply src/domains/sign/apim.tf]: Done!

**Review by manuelcoppotelli — COMMENTED** [review]
LGTM 👍

**Review by manuelcoppotelli — APPROVED** [review]

**Review by gunzip — APPROVED** [review]

**Pull request merged**